src/Layout/XmlLayoutBaseNS.cs

```diff
@@ -0,0 +1,281 @@
+#region Apache License
+//
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to you under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#endregion
+
+using System;
+using System.IO;
+using System.Text;
+using System.Xml;
+
+using log4net.Util;
+using log4net.Core;
+
+namespace log4net.Layout
+{
+	/// <summary>
+	/// Layout that formats the log events as XML elements using a namespace aware writer.
+	/// </summary>
+	/// <remarks>
+	/// <para>
+	/// This is an abstract class that must be subclassed by an implementation
+	/// to conform to a specific schema.
+	/// </para>
+	/// <para>
+	/// Deriving classes must implement the <see cref="FormatXml"/> method.
+	/// </para>
+	/// </remarks>
+	abstract public class XmlLayoutBaseNS : LayoutSkeleton
+	{
```

Comment:
Would you please explain the reasoning behind this base class? As far as I can tell we could reduce both the inheritance complexity and the public api without it. I discourage inheritance and encourage composition when there is need for it.

Reply:
Initially I planned to also create a replacement of `XmlLayout` as it suffers from the same problems if you specify a prefix. I'm not sure this is something people need anymore, in which case we can merge the classes.

Reply:
Does this indicate that we should rather implement a `XmlFormatter` class that can be plugged into various appenders? This is the composition what I have in mind and that we should prefer over inheritance.

Reply:
The layout is what gets plugged into the appenders, isn't it? The whole structure of `XmlLayoutBaseNS` and `XmlLayoutSchemaLog4jNS` just mimics what has been there before - an area of code I never really looked into so far.